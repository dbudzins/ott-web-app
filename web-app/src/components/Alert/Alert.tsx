import React from 'react';
import { useTranslation } from 'react-i18next';

import Dialog from '../Dialog/Dialog';
import Button from '../Button/Button';

import styles from './Alert.module.scss';

type Props = {
  open: boolean;
  message: string | null;
  onClose: () => void;
};

const Alert: React.FC<Props> = ({ open, message, onClose }: Props) => {
  const { t } = useTranslation('common');

  return (
    <Dialog open={open} onClose={onClose}>
      <h2 className={styles.title}>{t('alert.title')}</h2>
      <p className={styles.body}>{message}</p>
      <Button label={t('alert.close')} variant="outlined" onClick={onClose} fullWidth />
    </Dialog>
  );
};

export default Alert;
