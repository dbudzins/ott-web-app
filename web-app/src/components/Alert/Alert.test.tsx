import React from 'react';
import { render } from '@testing-library/react';

import Alert from './Alert';

vi.mock('../Dialog/Dialog', () => ({
  default: 'div',
}));

describe('<Alert>', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Alert message="Body" open={true} onClose={vi.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
