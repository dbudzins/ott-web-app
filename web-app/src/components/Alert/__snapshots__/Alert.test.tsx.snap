// Vitest Snapshot v1

exports[`<Alert> > renders and matches snapshot 1`] = `
<div>
  <div
    open=""
  >
    <h2
      class="title"
    >
      alert.title
    </h2>
    <p
      class="body"
    >
      Body
    </p>
    <button
      class="button default outlined fullWidth"
    >
      <span>
        alert.close
      </span>
    </button>
  </div>
</div>
`;
